
syntax = "proto3";

package DBMF;

service DBMFRPC{

  rpc RequestMultiFiles (MultiFilesRequest) returns (stream MutiFilesReply) {}

  rpc SayNewEpoch (NewEpochRequest) returns (NewEpochReply) {}

  rpc SynchNewEpoch (NewEpochRequest) returns (NewEpochReply) {}

}

message FileRequest {
  int32 request_idx=1;
  int32 worker_id=2;
}

message MultiFilesRequest {
  int32 node_id=1;
  repeated FileRequest file_request=2;
}

message FileReply {
  int32 reply_file_id=1;
  int32 worker_id=2;
  int32 request_idx=3;
  int32 file_size=4;
  int32 label_idx=5;
  bytes content=6;

}

message MutiFilesReply{
  int32 reply_file_num=1;
  repeated FileReply file_reply=2;
}

message NewEpochRequest {
  int32 node_id=1;
  int32 synch_id=2;
}

message NewEpochReply {
  int32 node_id=1;
  int32 node_status=2;
}


